mattermost.go

```diff
@@ -75,49 +77,94 @@ func main() {
 	}
 }
 
-func diagnosticsJob() {
+func securityAndDiagnosticsJob() {
 	go func() {
 		for {
-			if utils.Cfg.PrivacySettings.EnableDiagnostic && !model.IsOfficalBuild() {
+			if utils.Cfg.PrivacySettings.EnableSecurityFixAlert && model.IsOfficalBuild() {
 				if result := <-api.Srv.Store.System().Get(); result.Err == nil {
 					props := result.Data.(model.StringMap)
-					lastTime, _ := strconv.ParseInt(props["LastDiagnosticTime"], 10, 0)
+					lastSecurityTime, _ := strconv.ParseInt(props["LastSecurityTime"], 10, 0)
 					currentTime := model.GetMillis()
 
-					if (currentTime - lastTime) > 1000*60*60*24*7 {
-						l4g.Info("Sending error and diagnostic information to mattermost")
+					id := props["DiagnosticId"]
+					if len(id) == 0 {
+						id = model.NewId()
+						systemId := &model.System{Name: "DiagnosticId", Value: id}
+						<-api.Srv.Store.System().Save(systemId)
+					}
 
-						id := props["DiagnosticId"]
-						if len(id) == 0 {
-							id = model.NewId()
-							systemId := &model.System{Name: "DiagnosticId", Value: id}
-							<-api.Srv.Store.System().Save(systemId)
-						}
+					m := make(map[string]string)
+					m[utils.PROP_DIAGNOSTIC_ID] = id
+					m[utils.PROP_DIAGNOSTIC_BUILD] = model.CurrentVersion + "." + model.BuildNumber
+					m[utils.PROP_DIAGNOSTIC_DATABASE] = utils.Cfg.SqlSettings.DriverName
+					m[utils.PROP_DIAGNOSTIC_OS] = runtime.GOOS
+					m[utils.PROP_DIAGNOSTIC_CATEGORY] = utils.VAL_DIAGNOSTIC_CATEGORY_DEFALUT
+
+					if (currentTime - lastSecurityTime) > 1000*60*60*24*1 {
+						l4g.Info("Checking for security update from Mattermost")
 
-						systemLastTime := &model.System{Name: "LastDiagnosticTime", Value: strconv.FormatInt(currentTime, 10)}
-						if lastTime == 0 {
-							<-api.Srv.Store.System().Save(systemLastTime)
+						systemSecurityLastTime := &model.System{Name: "LastSecurityTime", Value: strconv.FormatInt(currentTime, 10)}
+						if lastSecurityTime == 0 {
+							<-api.Srv.Store.System().Save(systemSecurityLastTime)
 						} else {
-							<-api.Srv.Store.System().Update(systemLastTime)
+							<-api.Srv.Store.System().Update(systemSecurityLastTime)
 						}
 
-						m := make(map[string]string)
-						m[utils.PROP_DIAGNOSTIC_ID] = id
-						m[utils.PROP_DIAGNOSTIC_BUILD] = model.CurrentVersion + "." + model.BuildNumber
-						m[utils.PROP_DIAGNOSTIC_DATABASE] = utils.Cfg.SqlSettings.DriverName
-						m[utils.PROP_DIAGNOSTIC_OS] = runtime.GOOS
-						m[utils.PROP_DIAGNOSTIC_CATEGORY] = utils.VAL_DIAGNOSTIC_CATEGORY_DEFALUT
+						query := "?"
+						for name, value := range m {
+							if len(query) > 1 {
+								query += "&"
+							}
 
-						if ucr := <-api.Srv.Store.User().GetTotalUsersCount(); ucr.Err == nil {
-							m[utils.PROP_DIAGNOSTIC_USER_COUNT] = strconv.FormatInt(ucr.Data.(int64), 10)
+							query += name + "=" + utils.UrlEncode(value)
```

Comment:
Instead of making `m` a `map[string]string` you can use the `url.Values` object in the url package. It lets you add keys/values and has an `Encode()` function that will format all the keys/values into a query string for you

https://golang.org/pkg/net/url/#Values